packages/vscode-extension/src/webview/components/SendFilesOverlay.tsx

```diff
@@ -0,0 +1,158 @@
+import { VscodeProgressRing } from "@vscode-elements/react-elements";
+import "./Preview.css";
+import "./SendFilesOverlay.css";
+import { use$ } from "@legendapp/state/react";
+import { useProject } from "../providers/ProjectProvider";
+import classNames from "classnames";
+import { useSelectedDeviceSessionState } from "../hooks/selectedSession";
+
+const RETAIN_SUCCESS_SCREEN = 1000; // ms
+const RETAIN_ERROR_SCREEN = 3000; // ms
+
+// Important! You need to hold shift to drag files onto the panel
+// VSCode displays a "Hold shift to drop into editor" message when Preview is in the Editor Tab
+// but it doesn't show this when Preview is in the Side Panel
+export function SendFilesOverlay() {
+  const { project } = useProject();
+  const [isVisible, setIsVisible] = useState(false);
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("error");
+  const store$ = useSelectedDeviceSessionState();
+  const sendingFiles = use$(store$.sendingFiles);
+  const fileCount = sendingFiles.length;
+  const isLoading = fileCount > 0;
+
+  const resetOverlayState = useCallback(() => {
+    setIsSuccess(false);
+    setIsError(false);
+    setIsVisible(false);
+    setErrorMessage("");
+  }, []);
+
+  // Hide overlay after success and error animations
+  useEffect(() => {
+    if (isSuccess || isError) {
+      const timer = setTimeout(
+        resetOverlayState,
+        isSuccess ? RETAIN_SUCCESS_SCREEN : RETAIN_ERROR_SCREEN
+      );
+      return () => clearTimeout(timer);
+    }
+  }, [isSuccess, isError, resetOverlayState]);
+
+  const dragHandlers = useMemo(
+    () =>
+      ({
+        onDragEnter: (ev: React.DragEvent) => {
+          ev.preventDefault();
+          ev.stopPropagation();
+          if (isError) {
+            setIsError(false);
+            setErrorMessage("");
+          }
+          setIsVisible(true);
+        },
+        onDrop: async (ev: React.DragEvent) => {
+          ev.preventDefault();
+          ev.stopPropagation();
+          const files = ev.dataTransfer.files;
+
+          try {
+            const filePromises = [];
+            for (let i = 0; i < files.length; i++) {
+              const file = files[i];
+              const promise = file.arrayBuffer().then((buf) => {
+                return project.sendFileToDevice({
+                  fileName: file.name,
+                  data: buf,
+                });
+              });
+              filePromises.push(promise);
+            }
+
+            await Promise.all(filePromises);
+            setIsSuccess(true);
+          } catch (error) {
+            setIsError(true);
+            // Set a user-friendly error message
+            if (error instanceof Error && error.message) {
+              setErrorMessage(error.message);
+            } else {
+              setErrorMessage("Failed to send files. Please try again.");
+            }
```

Comment:
I think we should move the errors to the extension state as well, so that sending files via the device settings dropdown also displays the error/success overlays.